### DISCUSSION TITLE
How can I implement UIConfig in a monorepo?

### DISCUSSION THREAD

**Discussion by jmcmullen** [gluestack/gluestack-ui#1130] [Q&A]
I have a monorepo that looks like this:

```
node_modules
apps
  ├─ expo
  └─ next.js
packages
  └─ app
      └─ gluestack-ui.config.ts
```

The app directory shares UI code with expo and nextjs. I am using pnpm to install which is placing @gluestack-style/react inside the root node_modules. When I try to override the theme config, typescript doesn't detect the extended properties on UIConfig. I suspect this is because the override is looking in the wrong place? What's the best way to proceed?

**Comment by topkat** [comment]
By the way, have you make your monorepo functional ? I am struggeling just to setup a monorepo, I have used the last starter template from gluestack, so any advice may be welcome.

**Reply by liuqiaowei512** [reply]
@topkat any luck setting up the monorepo with gluestack? I tried using `npx gluestack-ui init --projectType library` but no luck. In my case, I have a monorepo with two expo apps, and I want to place gluestack-ui in a shared package

**Reply by topkat** [reply]
@liuqiaowei512 not actually...I'm still looking forward for a solution.
